### PR TITLE
Add content to docs/perf

### PR DESCRIPTION
**Pull request by liyuqian**
This is an attempt to port the content from go/flutter-performance doc
to our public pages.

### CODE REVIEW

**Comment by liyuqian** [comment]
@googlebot I consent.

**Review by sfshaza2 — COMMENTED** [review]
Manually changed PR to cla:yes, which @liyuqian approved. I am still in the middle of my review.

File: src/docs/perf/appendix.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+In this context, performance isn't the execution of an action itself; 
+it’s how well something or someone performs. Therefore, we use the adjective
+ _performant_.
+
+While the _how_ _well_ part can, in general, be described in natural languages, 
+in our limited scope, the focus is on something that is quantifiable as a real
+number. Real numbers include integers and 0/1 binaries as special cases.
+Natural language descriptions are still very important. For example, a news
+article that heavily criticizes Flutter’s performance by just using words
+without any numbers (a quantifiable value) could still be devastating. The
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Yes, your understanding is correct. I've modified it to "could still be meaningful, and it
could have great impacts". Is that better?

File: src/docs/perf/appendix.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+## What is performance?
+
+Performance is a set of quantifiable properties of a performer.
+
+In this context, performance isn't the execution of an action itself; 
+it’s how well something or someone performs. Therefore, we use the adjective
+ _performant_.
+
+While the _how_ _well_ part can, in general, be described in natural languages,
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/appendix.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+they heavily affect how fast Flutter can incorporate code changes, to improve 
+the app's performance. Flutter team performance has the engineers inside the 
+Flutter team as performers. Flutter community performance has the open source 
+community as the performer, and the performance metric could be the median time 
+to close a community GitHub pull request.
+
+Here, the scope was intentionally broadened to include performance issues other
+than just app performance issues because they can share many tools regardless of
+who the performers are. For example, Flutter app performance and infra
+performance might share the same dashboard and similar alert mechanisms.
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply src/docs/perf/appendix.md:38]: Shortened. Please let me know if it looks better now.

File: src/docs/perf/appendix.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+variable, or you could take the average of many measurements to eliminate the
+noise by [the law of large numbers][1].
+
+### 3. Performance is comparable and convertible
+
+Performance numbers not only have unambiguous meanings, but they also have
+unambiguous comparisons. For example, there’s no doubt that 5 is greater than 4.
+On the other hand, it might be subjective to figure out whether excellent is
+better or worse than superb. Similarly, could you figure out whether epic is
+better than legendary? Actually, the phrase _strongly_ _exceeds_ _expectations_
```
Comment (sfshaza2) [inline src/docs/perf/appendix.md:106]: You can underline whole phrases, so =>

```
_strongly exceeds expectations_
```

File: src/docs/perf/appendix.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+who are closer to the development team, who have a larger bandwidth and lower
+cost for chatting or face-to-face meetings.
+
+By having the same metrics to detect problems no matter how far away or how
+silent the users are, we can treat all issues fairly. That, in turn,
+allows us to focus on the right issues that have greater effects.
+
+### How to make performance useful
+
+The following is a summary the 4 points that were discussed, from a slightly
```
Comment (sfshaza2) [inline src/docs/perf/appendix.md:133]: => The following summarizes the 4 points discussed here, from a slightly different perspective.  (Note that perspective was misspelled.)

File: src/docs/perf/appendix.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+different perspctive:
+1. Make performance metrics easy to consume. Do not overwhelm the readers with a
+   lot of numbers (or words). If there are many numbers, then try to summarize
+   them into a smaller set of numbers (for exmaple, summarize many numbers into
```
Comment (sfshaza2) [inline src/docs/perf/appendix.md:137]: exmaple => example

File: src/docs/perf/appendix.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+### 4. Performance is fair
+
+If issues rely on verbose words to be discovered, then an unfair advantage is
+given to people who are more verbose (more willing to chat or write) or those
+who are closer to the development team, who have a larger bandwidth and lower
+cost for chatting or face-to-face meetings.
+
+By having the same metrics to detect problems no matter how far away or how
+silent the users are, we can treat all issues fairly. That, in turn,
+allows us to focus on the right issues that have greater effects.
```
Comment (sfshaza2) [inline src/docs/perf/appendix.md:129]: effects => impact

File: src/docs/perf/faq.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  - If you use IntelliJ, a GUI view of this data is available. Select
+    **show widget rebuild information**, and you will see which widgets
+    rebuild visually in your IDE.
+
+- How do I query the target frames per second (of the display)?
+  - https://github.com/flutter/flutter/wiki/Engine-specific-Service-Protocol-extensions#get-the-display-refresh-rate-_fluttergetdisplayrefreshrate
+
+- How to solve my app’s poor animations caused by an expensive Dart async
+  function call that is blocking the UI thread?
+  - Spawn another isoalte using
```
Comment (sfshaza2) [inline src/docs/perf/faq.md:51]: isolte => isolate

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
I have a few minor nits but, overall, this looks good.

File: src/docs/perf/appendix.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+Another advantage of having performance as a set of numbers is its unambiguity.
+When you want an animation to have a performance of 20 ms per frame or
+50 fps, there’s little room for different interpretations about the numbers. On
+the other hand, to describe the same animation in words, someone might call it
+good, while someone else might complain that it’s bad. Similarly, the same
+word or phrase could be interpreted differently by different people. You might
+interpret an OK frame rate to be 60 fps, while someone else might interpret it
+to be 30 fps.
+
+Numbers could still be noisy. For example, the measured time per frame might
```
[the remark this answers is not in the document]
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/appendix.md (hunk 4 above)
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/appendix.md (hunk 7 above)
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/appendix.md (hunk 5 above)
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/appendix.md (hunk 6 above)
Comment (liyuqian) [reply src/docs/perf/appendix.md]: Done.

File: src/docs/perf/faq.md (hunk 8 above)
Comment (liyuqian) [reply src/docs/perf/faq.md]: Done.

**Review by liyuqian — COMMENTED** [review]
Fixed :)

**Review by sfshaza2 — APPROVED** [review]
LGTM

**Pull request merged**